### PR TITLE
Fix memory cache

### LINKED ISSUE #761: [REGRESSION] Implicit Tiling Repeating request
**Description**

Not exactly the same as https://github.com/NASA-AMMOS/3DTilesRendererJS/issues/728
I've noticed that after moving the camera around, we start doing the same requests again and again, infinitely.

Could be link to the latest changes concerning the cache management, WDYT ?

**To Reproduce**

1. Load the tileset from https://github.com/AnthonyGlt/data/tree/main/arbres_rieux_glb
2. Rotate / Move the camera around
3. Check the network request

[Screencast from 23-09-2024 17:59:56.webm](https://github.com/user-attachments/assets/adc05089-b3fc-403f-937c-5fd184d9e2c3)

### PR DESCRIPTION
**Pull request by gkjohnson**
Fix #761

### CODE REVIEW

**Comment by gkjohnson** [comment]
cc @AnthonyGlt - this should be ready if you want to test it a bit. I just have to write some more tests / fix the existing ones.

In simpler terms, the changes are:
- Split the "max size" eviction and the "min size" eviction while loops so the logic is easier to follow.
- Add a concept of "fully loaded" items. If an item isn't full loaded then it can be evicted from the cache if it's above the "max" size even if its marked as "used". This lets us stop any downloading or yet-to-be-parsed tile geometry if the cache has suddenly become full.
- The loading logic now checks the if the geometry memory usage has been assigned before determining if tile data should be evicted after loading.

**Comment by gkjohnson** [comment]
PR should be ready, now.

File: src/utilities/LRUCache.js [hunk 1]
```diff
@@ -135,6 +145,28 @@ class LRUCache {
 
 	}
 
+	// Marks whether tiles in the cache have been completely loaded or not. Tiles that have not been completely
+	// loaded are subject to being disposed early even if the are marked as used if the cache is full above its
+	// max size limits.
```
Comment (AnthonyGlt) [inline src/utilities/LRUCache.js:150]: nit There is a word missing here

Comment (AnthonyGlt) [reply src/utilities/LRUCache.js:150]: the test check is a good addition :+1:

Comment (gkjohnson) [reply src/utilities/LRUCache.js]: Thanks - just reworded it. Let me know once you think everything looks good and we can get this merged and a new version published.

**Review by AnthonyGlt — APPROVED** [review]
Looks good to me :+1: 
We got to juggle with the notions of `used` and `loaded` but at the end, it makes the cache management more subtle.
Well done :D

**Comment by gkjohnson** [comment]
Thanks - yeah nomenclature could probably use some work but I think the functionality is good.
